src/components/Deck.js

```diff
@@ -65,11 +65,13 @@ const Deck = ({ children, loop, keyboardControls }) => {
     loop ? true : false
   );
 
-  const transitions = useTransition(
-    state.currentSlide,
-    p => p,
-    filteredChildren[state.currentSlide].props.transitionEffect || defaultSlideEffect
-  );
+  const transitions = useTransition(state.currentSlide, p => p, {
+    ...(filteredChildren[state.currentSlide].props.transitionEffect ||
+      defaultSlideEffect),
+    config: { precision: 0 },
+    unique: true,
+    immediate: rest.animationsWhenGoingBack ? false : state.immediate
```

Comment:
This is a nit picky comment but could we pass `animationsWhenGoingBack` as an argument to `useDeck`? It might be nice to put all the "immediate" logic in one place. Not a big deal though, feel free to ignore.

Reply:
Yeaaah this is a great suggestion!